parsing/lexer.mll

```diff
@@ -576,8 +578,10 @@ rule token = parse
   | '#' symbolchar_or_hash + as op
             { HASHOP op }
   | "let" kwdopchar dotsymbolchar * as op
+  | "let" kwdopchar? '.' letdotopsymbolchar * as op
```

Comment:
Why not extending `letdotopsymbolchar` to include uppercase and saying only the first one must be a initial one? Yours approach would not allow for `let.some.map` for example.

But the current one allow for some weird stuff such as `let.some.....`
```
| "let" kwdopchar? '.' lowercase letdotopsymbolchar* as op
```

Reply:
The token `.` is generally used for projection or paths, typically I would expect `let.some.map` and `let.some.bind` to be somehow related to a `let.some` but that is not the case here. 

It seems fine to me to be conservative and first limit ourselves to the syntax that we really want like `let.option`, `let.some_map`, `let.ok`, `let.error_map`, or `let.app_list` and leave some rooms for future options. After all, it is far easier to relax the syntax constraint in the future compared to trying to deprecate an existing but nearly totally unused syntax.

Reply:
In case additional opinions can help: I agree that I prefer to allow `let*.some` but rule out `let.some*` for now (so: keeping operator symbols adjacent to the keyword, not in other places), but I would be fine with allowing `let.some.map` as well. It is probably wise to disallow Uidents for now, to leave room for a proper qualification semantics in the future, and generally to try to under-approximate rather than over-approximate the allowed utterances, to keep freedom for future extensions.

Reply:
I made the changes as they do make sense and I will make the discussion worse.

The usage of `.` is probably bad, as `.` commonly has the notion of accessing something.

I know that the analogy here is that `let.some` means `access the Some and do some stuff inside of it`. But for that meaning as a language designer I would much rather allow only `let*.some`. And in the future I would do the breaking change and not add the `.`, so `let*some`, maybe add a lexer warning?

But I know that we're being victim's of the [Wadler's Law](https://wiki.haskell.org/Wadler's_Law), and as an OCaml developer, I think `let.some` is way better than `let*.some`, even thought it is a single token. It just reads naturally.

Worst case scenario we have even another bit of syntax that is half broken and that at some point will need a deprecation policy.

Reply:
Is `let.some` intended to be the `map` or the `bind` for the option type? I'm not sure.

Reply:
That's a convention up to the user, in the same way that `* | +` are conventions, I personally don't ever add `map` so `let.some` is always bind.

My usage is a pairing of `let.some` for `bind` and `some` for `return`.

Reply:
`let.some x = 1 in some (x + 2)`